### DISCUSSION TITLE
Implement NavigationStack

### DISCUSSION THREAD

**Discussion by kefahB** [NativeScript/NativeScript#10112] [Ideas]
Hi @NathanWalker 

It will be useful to keep the 10 last `Frame.navigationStackEntry: NavigationEntry[]` exposed to us, this can be handy for duplicated navigation scenario such as navigation when receive a notification, if we have `navigationStackEntry` then it is simple to check if we are in the same page.

Each time we call `Frame.navigate()` we push the entry on the navigationStackEntry.

**Comment by shirakaba** [comment]
This seems like a user-specific use-case, so couldn't this be implemented in userland instead of Core?

(If the properties are private, then fair enough, but I don't think holding onto an arbitrary history stack is a good idea in terms of memory management).

**Comment by kefahB** [comment]
I agree with you but it is a lot off stuff to do that each time we call navigate! I am using Core and if we haven't implemented a router we couldn't handle it.

I think it is a good think if NS propose a variety of helpers, I see that positive :)
